src/nn/webgl/TfjsModel.js

```diff
@@ -511,31 +551,49 @@ export default class WebGLModel {
           filterW = operands[inputs[i++]].value[0];
           filterH = operands[inputs[i++]].value[0];
           activation = FuseFunctionMap.get(operands[inputs[i++]].value[0]);
-          if (this._isPaddingEqual(paddingLeft, paddingRight,
-                                   paddingTop, paddingBottom)) {
-            if (op === OperationCode.AVERAGE_POOL_2D) {
-              output.assign(activation(
-                  input.avgPool([filterH, filterW],
-                                [strideH, strideW],
-                                paddingLeft, 'floor')));
-            } else {
-              output.assign(activation(
-                  input.maxPool([filterH, filterW],
-                                [strideH, strideW],
-                                paddingLeft, 'floor')));
+          if (this._model._backend==='WASM' && filterH===1 && filterW===1) {
+            let inputData = input.pad([[0, 0], [paddingTop, paddingBottom],
+                                               [paddingLeft, paddingRight], [0, 0]]).dataSync();            
+            const [N, H, W, C] = output.shape;
+            const [NI, HI, WI, CI] = input.shape;
+            let outputData = new Float32Array(N*H*W*C);
+             for (let n = 0; n < N; ++n) {
+              for (let h = 0; h < H; ++h) {
+                for (let w = 0; w < W; ++w) {
+                  for (let c = 0; c < C; ++c) {
+                    outputData[n*H*W*C + h*W*C + w*C + c] = inputData[n*HI*WI*CI + h*strideH*WI*CI + w*strideW*C + c];
+                  }
+                }
+              }
```

Comment:
can we make a function to avoid duplicating code with above occurance?

Reply:
Okay, I'll do that.